Utils/Extensions/NSURLExtensions.swift

```diff
@@ -96,6 +96,14 @@ extension NSURL {
         }
         return nil
     }
+    
+    public func baseDomainAndPath() -> String? {
+        if let baseDomain = self.baseDomain() {
+            return self.baseDomain()! + (self.path ?? "/")
+        } else {
+            return nil
```

Comment:
No need for the `else`; you're returning from the `if` block. Just `return nil`.